src/utils/registry.ts

```diff
@@ -0,0 +1,272 @@
+    'mac10.15': '%s/builds/firefox/%s/firefox-mac-10.14.zip',
+    'mac11': '%s/builds/firefox/%s/firefox-mac-10.14.zip',
+    'mac11-arm64': '%s/builds/firefox/%s/firefox-mac-11.0-arm64.zip',
+    'win32': '%s/builds/firefox/%s/firefox-win32.zip',
+    'win64': '%s/builds/firefox/%s/firefox-win64.zip',
+  },
+  webkit: {
+    'ubuntu18.04': '%s/builds/webkit/%s/webkit-ubuntu-18.04.zip',
+    'ubuntu20.04': '%s/builds/webkit/%s/webkit-ubuntu-20.04.zip',
+    'mac10.13': undefined,
+    'mac10.14': '%s/builds/webkit/%s/webkit-mac-10.14.zip',
+    'mac10.15': '%s/builds/webkit/%s/webkit-mac-10.15.zip',
+    'mac11': '%s/builds/webkit/%s/webkit-mac-10.15.zip',
+    'mac11-arm64': '%s/builds/webkit/%s/webkit-mac-11.0-arm64.zip',
+    'win32': '%s/builds/webkit/%s/webkit-win64.zip',
+    'win64': '%s/builds/webkit/%s/webkit-win64.zip',
+  },
+  ffmpeg: {
+    'ubuntu18.04': '%s/builds/ffmpeg/%s/ffmpeg-linux.zip',
+    'ubuntu20.04': '%s/builds/ffmpeg/%s/ffmpeg-linux.zip',
+    'mac10.13': '%s/builds/ffmpeg/%s/ffmpeg-mac.zip',
+    'mac10.14': '%s/builds/ffmpeg/%s/ffmpeg-mac.zip',
+    'mac10.15': '%s/builds/ffmpeg/%s/ffmpeg-mac.zip',
+    'mac11': '%s/builds/ffmpeg/%s/ffmpeg-mac.zip',
+    'mac11-arm64': '%s/builds/ffmpeg/%s/ffmpeg-mac.zip',
+    'win32': '%s/builds/ffmpeg/%s/ffmpeg-win32.zip',
+    'win64': '%s/builds/ffmpeg/%s/ffmpeg-win64.zip',
+  },
+};
+
+export const allBrowserNames: BrowserName[] = Object.keys(DOWNLOAD_URLS) as BrowserName[];
+
+export const hostPlatform = ((): BrowserPlatform => {
+  const platform = os.platform();
+  if (platform === 'darwin') {
+    const [major, minor] = execSync('sw_vers -productVersion', {
+      stdio: ['ignore', 'pipe', 'ignore']
+    }).toString('utf8').trim().split('.').map(x => parseInt(x, 10));
+    let arm64 = false;
+    // BigSur is the first version that might run on Apple Silicon.
+    if (major >= 11) {
+      arm64 = execSync('sysctl -in hw.optional.arm64', {
+        stdio: ['ignore', 'pipe', 'ignore']
+      }).toString().trim() === '1';
+    }
+    // We do not want to differentiate between minor big sur releases
+    // since they don't change core APIs so far.
+    const macVersion = major === 10 ? `${major}.${minor}` : `${major}`;
+    const archSuffix = arm64 ? '-arm64' : '';
+    return `mac${macVersion}${archSuffix}` as BrowserPlatform;
+  }
+  if (platform === 'linux') {
+    const ubuntuVersion = getUbuntuVersionSync();
+    if (parseInt(ubuntuVersion, 10) <= 19)
+      return 'ubuntu18.04';
+    return 'ubuntu20.04';
+  }
+  if (platform === 'win32')
+    return os.arch() === 'x64' ? 'win64' : 'win32';
+  return platform as BrowserPlatform;
+})();
+
+export const registryDirectory = (() => {
+  const envDefined = getFromENV('PLAYWRIGHT_BROWSERS_PATH');
+  if (envDefined === '0')
+    return path.join(__dirname, '..', '..', '.local-browsers');
+  if (envDefined)
+    return path.resolve(process.cwd(), envDefined);
+  return path.join(cacheDirectory(), 'ms-playwright');
+})();
```

Comment:
Let's always do `path.resolve` in this method, to hopefully fix https://github.com/microsoft/playwright/issues/4867?

Reply:
done